Index: hibiscus/src/de/willuhn/jameica/hbci/passports/ddv/DDVConfigFactory.java
===================================================================
--- hibiscus.orig/src/de/willuhn/jameica/hbci/passports/ddv/DDVConfigFactory.java
+++ hibiscus/src/de/willuhn/jameica/hbci/passports/ddv/DDVConfigFactory.java
@@ -530,9 +530,9 @@ public class DDVConfigFactory
    */
   private static File getJNILib() throws ApplicationException
   {
-    String file = null;
+    String file = "libhbci4java-card-linux.so";
     
-    switch (Application.getPlatform().getOS())
+    /*switch (Application.getPlatform().getOS())
     {
       case Platform.OS_LINUX:
         file = "libhbci4java-card-linux-32.so";
@@ -561,12 +561,12 @@ public class DDVConfigFactory
       case Platform.OS_FREEBSD_64:
         file = "libhbci4java-card-freebsd-64.so";
         break;
-    }
+    }*/
     
     if (file == null)
       throw new ApplicationException(i18n.tr("Hibiscus unterstützt leider keine Chipkartenleser für Ihr Betriebssystem"));
 
-    File f = new File(de.willuhn.jameica.hbci.Settings.getLibPath(),file);
+    File f = new File("/usr/lib",file);
     if (!f.exists())
       throw new ApplicationException(i18n.tr("Treiber {0} nicht gefunden",f.getAbsolutePath()));
 
@@ -612,4 +612,4 @@ public class DDVConfigFactory
  * Revision 1.1  2010-09-07 15:28:05  willuhn
  * @N BUGZILLA 391 - Kartenleser-Konfiguration komplett umgebaut. Damit lassen sich jetzt beliebig viele Kartenleser und Konfigurationen parellel einrichten
  *
- **********************************************************************/
\ No newline at end of file
+ **********************************************************************/
