Index: jameica.scripting/src/de/willuhn/jameica/scripting/services/impl/ScriptingServiceImpl.java
===================================================================
--- jameica.scripting.orig/src/de/willuhn/jameica/scripting/services/impl/ScriptingServiceImpl.java
+++ jameica.scripting/src/de/willuhn/jameica/scripting/services/impl/ScriptingServiceImpl.java
@@ -89,6 +89,11 @@ public class ScriptingServiceImpl implem
     // 1. Script-Engine laden
     ScriptEngineManager factory = new ScriptEngineManager();
     this.engine = factory.getEngineByName("JavaScript");
+
+    // Some JREs do not necessarily provide a JavaScript engine
+    if (this.engine == null)
+        throw new RuntimeException("Your Java Runtime Environment does not provide a JavaScript engine. Jameica Scripting won't work");
+
     this.engine.put("events",this.events);
 
     // 2. Vom Benutzer registrierte Scripts ausfuehren.
@@ -273,4 +278,4 @@ public class ScriptingServiceImpl implem
  * Revision 1.1  2010-07-23 12:58:33  willuhn
  * @N initial import
  *
- **********************************************************************/
\ No newline at end of file
+ **********************************************************************/
