Index: willuhn-hbci4java-9999/chipcard/src/frontend/Makefile
===================================================================
--- willuhn-hbci4java-9999.orig/chipcard/src/frontend/Makefile
+++ willuhn-hbci4java-9999/chipcard/src/frontend/Makefile
@@ -1,4 +1,3 @@
-JAVA_HOME = $(shell readlink -f /usr/bin/javac | sed "s:/bin/javac::")
 DEFINES   = -D__UNIX
 INCLUDES  = -I../include -I$(JAVA_HOME)/include -I$(JAVA_HOME)/include/linux
 WARNINGS  = -Wall
