Index: hbci4java-2.5.12-src/chipcard/src/frontend/Makefile
===================================================================
--- hbci4java-2.5.12-src.orig/chipcard/src/frontend/Makefile
+++ hbci4java-2.5.12-src/chipcard/src/frontend/Makefile
@@ -4,8 +4,9 @@ INCLUDES = -I../include -I$(JAVA_HOME)/i
 WARNINGS = -Wall
 # removed "-Werror -pedantic" because of dlsym() typecast problem
 
-CC       = g++
-CFLAGS   = -fPIC $(DEFINES) $(INCLUDES) $(WARNINGS)
+CXX      ?= g++
+AR       ?= ar
+CXXFLAGS += -fPIC $(DEFINES) $(INCLUDES) $(WARNINGS)
 LIBS     = -L../../bin -lzkachip-highlevel -lzkachip-lowlevel
 
 SOURCES = frontend.cpp
@@ -15,16 +16,16 @@ OBJECTS = ../../bin/frontend.o
 .SUFFIXES: .cpp .o
 
 ../../bin/%.o: %.cpp
-	$(CC) -c -o $@ $(CFLAGS) $<
+	$(CXX) -c -o $@ $(CXXFLAGS) $(LDFLAGS) $<
 
 all: depend ../../bin/libhbci4java-card-linux.so
 
 depend: Makefile.depend
 
 Makefile.depend: $(SOURCES)
-	for file in $(SOURCES); do echo -n "../../bin/"; gcc $(CFLAGS) -MM $$file; done >Makefile.depend
+	for file in $(SOURCES); do echo -n "../../bin/"; $(CXX) $(CXXFLAGS) $(LDFLAGS) -MM $$file; done >Makefile.depend
 
 ../../bin/libhbci4java-card-linux.so: $(OBJECTS) ../../bin/libzkachip-highlevel.a ../../bin/libzkachip-lowlevel.a
-	$(CC) -fPIC -shared -o $@ $(OBJECTS) $(LIBS) 
+	$(CXX) $(LDFLAGS) -fPIC -shared -o $@ $(OBJECTS) $(LIBS) 
 
 include Makefile.depend
Index: hbci4java-2.5.12-src/chipcard/src/highlevel/Makefile
===================================================================
--- hbci4java-2.5.12-src.orig/chipcard/src/highlevel/Makefile
+++ hbci4java-2.5.12-src/chipcard/src/highlevel/Makefile
@@ -4,9 +4,9 @@ INCLUDES = -I../include
 WARNINGS = -Wall
 # removed "-Werror -pedantic" because of dlsym() typecast problem
 
-CC       = g++
-AR       = ar
-CFLAGS   = -fPIC $(DEFINES) $(INCLUDES) $(WARNINGS)
+CXX       ?= g++
+AR        ?= ar
+CXXFLAGS  += -fPIC $(DEFINES) $(INCLUDES) $(WARNINGS)
 
 SOURCES = hbci_cardtype.cpp \
           ddv_decrypt.cpp \
@@ -37,14 +37,14 @@ OBJECTS = ../../bin/hbci_cardtype.o \
 .SUFFIXES: .cpp .o
 
 ../../bin/%.o: %.cpp
-	$(CC) -c -o $@ $(CFLAGS) $<
+	$(CXX) -c -o $@ $(CXXFLAGS) $(LDFLAGS) $<
 
 all: depend ../../bin/libzkachip-highlevel.a
 
 depend: Makefile.depend
 
 Makefile.depend: $(SOURCES)
-	for file in $(SOURCES); do echo -n "../../bin/"; gcc $(CFLAGS) -MM $$file; done >Makefile.depend
+	for file in $(SOURCES); do echo -n "../../bin/"; $(CXX) $(CXXFLAGS) $(LDFLAGS) -MM $$file; done >Makefile.depend
 
 ../../bin/libzkachip-highlevel.a: $(OBJECTS)
 	$(AR) rcsv $@ $(OBJECTS)
Index: hbci4java-2.5.12-src/chipcard/src/lowlevel/Makefile
===================================================================
--- hbci4java-2.5.12-src.orig/chipcard/src/lowlevel/Makefile
+++ hbci4java-2.5.12-src/chipcard/src/lowlevel/Makefile
@@ -4,9 +4,9 @@ INCLUDES = -I../include
 WARNINGS = -Wall
 # removed "-Werror -pedantic" because of dlsym() typecast problem
 
-CC       = g++
-AR       = ar
-CFLAGS   = -fPIC $(DEFINES) $(INCLUDES) $(WARNINGS)
+CXX      ?= g++
+AR       ?= ar
+CXXFLAGS += -fPIC $(DEFINES) $(INCLUDES) $(WARNINGS)
 
 SOURCES = atr.cpp \
           bcs.cpp \
@@ -41,14 +41,14 @@ OBJECTS = ../../bin/atr.o \
 .SUFFIXES: .cpp .o
 
 ../../bin/%.o: %.cpp
-	$(CC) -c -o $@ $(CFLAGS) $<
+	$(CXX) -c -o $@ $(CXXFLAGS) $(LDFLAGS) $<
 
 all: depend ../../bin/libzkachip-lowlevel.a
 
 depend: Makefile.depend
 
 Makefile.depend: $(SOURCES)
-	for file in $(SOURCES); do echo -n "../../bin/"; gcc $(CFLAGS) -MM $$file; done >Makefile.depend
+	for file in $(SOURCES); do echo -n "../../bin/"; $(CXX) $(CXXFLAGS) $(LDFLAGS) -MM $$file; done >Makefile.depend
 
 ../../bin/libzkachip-lowlevel.a: $(OBJECTS)
 	$(AR) rcsv $@ $(OBJECTS)
Index: hbci4java-2.5.12-src/chipcard/src/tools/Makefile
===================================================================
--- hbci4java-2.5.12-src.orig/chipcard/src/tools/Makefile
+++ hbci4java-2.5.12-src/chipcard/src/tools/Makefile
@@ -4,8 +4,9 @@ INCLUDES = -I../include
 WARNINGS = -Wall
 # removed "-Werror -pedantic" because of dlsym() typecast problem
 
-CC       = g++
-CFLAGS   = -fPIC $(DEFINES) $(INCLUDES) $(WARNINGS)
+CXX      ?= g++
+AR       ?= ar
+CXXFLAGS += -fPIC $(DEFINES) $(INCLUDES) $(WARNINGS)
 LIBS	 = -L../../bin -lzkachip-highlevel -lzkachip-lowlevel -ldl
 
 SOURCES = t_hbci_cardinfo.cpp \
@@ -29,15 +30,15 @@ PROGRAMS = ../../bin/t_hbci_cardinfo \
 .SUFFIXES: .cpp .o
 
 ../../bin/%.o: %.cpp
-	$(CC) -c -o $@ $(CFLAGS) $<
+	$(CXX) -c -o $@ $(CXXFLAGS) $(LDFLAGS) $<
 ../../bin/%: ../../bin/%.o ../../bin/libzkachip-highlevel.a ../../bin/libzkachip-lowlevel.a
-	$(CC) -fPIC -o $@ $< $(LIBS)
+	$(CXX) -fPIC -o $@ $< $(LIBS)
 
 all: depend $(PROGRAMS)
 
 depend: Makefile.depend
 
 Makefile.depend: $(SOURCES)
-	for file in $(SOURCES); do echo -n "../../bin/"; gcc $(CFLAGS) -MM $$file; done >Makefile.depend
+	for file in $(SOURCES); do echo -n "../../bin/"; $(CXX) $(CXXFLAGS) $(LDFLAGS) -MM $$file; done >Makefile.depend
 
 include Makefile.depend
